extensions/amp-story-shopping/0.1/amp-story-shopping-tag.css

```diff
@@ -51,38 +54,178 @@ amp-story-shopping-tag {
 }
 
 .i-amphtml-amp-story-shopping-tag-dot {
-  width: 16px !important;
-  height: 16px !important;
+  --i-amphtml-tag-dot-size: 16px !important;
+  width: var(--i-amphtml-tag-dot-size) !important;
+  height: var(--i-amphtml-tag-dot-size) !important;
   border-radius: 100% !important;
-  background-color: var(--i-amphtml-shopping-tag-bg-color) !important;
   position: relative !important;
   display: flex !important;
   align-items: center !important;
   justify-content: center !important;
-  margin-inline-end: 4px !important;
-  flex-shrink: 0 !important;
-  backdrop-filter: var(--i-amphtml-shopping-tag-backdrop-filter) !important;
+  margin: 0 4px !important;
   flex-shrink: 0 !important;
 }
 
 .i-amphtml-amp-story-shopping-tag-dot:before {
-  border-radius: 100% !important;
   content: '' !important;
-  width: 8px !important;
-  height: 8px !important;
-  background: white !important;
-  box-shadow: 0px 2px 8px var(--i-amphtml-shopping-tag-bg-color) !important;
+  position: absolute !important;
+  border-radius: inherit !important;
+  width: var(--i-amphtml-tag-dot-size) !important;
+  height: var(--i-amphtml-tag-dot-size) !important;
+  background-color: var(--i-amphtml-shopping-tag-bg-color) !important;
+}
+
+[active] .i-amphtml-amp-story-shopping-tag-dot:before {
```

Comment:
Is it possible that there is – or will eventually be – an ancestor element other than the shopping tag itself that has an attribute called 'active'? If so, then it seems like many of your new selectors can unintentionally be triggered by random elements